Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+    if (!s_dependencyPropertiesRegistered) {
+        s_dependencyPropertiesRegistered = true;
+
+        s_textProperty = DependencyProperty::RegisterAttached("Text",
+            Platform::String::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_secureEntryProperty = DependencyProperty::RegisterAttached("SecureEntry",
+            Platform::Boolean::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_placeholderProperty = DependencyProperty::RegisterAttached("Placeholder",
+            Platform::String::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_inputScopeProperty = DependencyProperty::RegisterAttached("InputScope",
+            Input::InputScope::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_enabledProperty = DependencyProperty::RegisterAttached("Enabled",
+            Platform::Boolean::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_foregroundProperty = DependencyProperty::RegisterAttached("Foreground",
+            Media::SolidColorBrush::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_textAlignmentProperty = DependencyProperty::RegisterAttached("TextAlignment",
+            Windows::UI::Xaml::TextAlignment::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_textVerticalAlignmentProperty = DependencyProperty::RegisterAttached("TextVerticalAlignment",
+            Windows::UI::Xaml::VerticalAlignment::typeid,
+            TextField::typeid,
+            nullptr);
+
+        s_borderStyleProperty = DependencyProperty::RegisterAttached("BorderStyle",
+            UIKit::Xaml::TextBorderStyle::typeid,
+            TextField::typeid,
+            nullptr);
+    }
+}
+
+// Accessor for our Layer content
+Image^ TextField::LayerContent::get() {
+    return nullptr;
+}
+
+// Accessor for our Layer content
+bool TextField::HasLayerContent::get() {
+    return _content != nullptr;
+}
+
+// Accessor for our SublayerCanvas
+Canvas^ TextField::SublayerCanvas::get() {
+    return nullptr;
+}
+
+// Accessor for the LayerProperty that manages the BorderBrush of this TextField
+Private::CoreAnimation::LayerProperty^ TextField::GetBorderBrushProperty() {
+    return nullptr;
+}
+
+// Accessor for the LayerProperty that manages the BorderThickness of this TextField
+Private::CoreAnimation::LayerProperty^ TextField::GetBorderThicknessProperty() {
+    return nullptr;
+}
+
+TextBox^ TextField::TextBox::get() {
+    return safe_cast<Windows::UI::Xaml::Controls::TextBox^>(_backingControl);
+}
+
+PasswordBox^ TextField::PasswordBox::get() {
+    return safe_cast<Windows::UI::Xaml::Controls::PasswordBox^>(_backingControl);
+}
+
+void TextField::KillFocus() {
+    if (SecureEntry) {
+        PasswordBox->IsEnabled = false;
+        PasswordBox->IsEnabled = true;
+    } else {
+        TextBox->IsEnabled = false;
+        TextBox->IsEnabled = true;
+    }
```

Comment:
should add a comment about how this is a hack and we should switch to Windows APIs if/when available